services/cannon/src/Cannon/Run.hs

```diff
@@ -87,10 +83,8 @@ run o = do
       server =
         hoistServer (Proxy @PublicAPI) (runCannonToServant e) publicAPIServer
           :<|> hoistServer (Proxy @Internal.API) (runCannonToServant e) internalServer
-  tid <- myThreadId
-  void $ installHandler sigTERM (signalHandler (env e) tid) Nothing
-  void $ installHandler sigINT (signalHandler (env e) tid) Nothing
-  runSettings s app `finally` do
+  runSettingsWithShutdown s app 5 `finally` do
```

Comment:
I am not sure about the impact of this 5 second wait, so I will run a couple of experiments to see that the draining happens correctly. Earlier we were also not closing the listen socket, so impact of that is another mystery to me.
I am being super careful because this functionality is important and we don't have nice ways of automatically testing it.

Reply:
67946bd37a4354cfff512b35d761d6fdf8433360 (also 0f264d2eaec2b709d86820620f892f169157d286)

Reply:
> [67946bd](https://github.com/wireapp/wire-server/commit/67946bd37a4354cfff512b35d761d6fdf8433360) (also [0f264d2](https://github.com/wireapp/wire-server/commit/0f264d2eaec2b709d86820620f892f169157d286))

no, that's different commits.  but the github UI is messing with me, so i'll just leave you to the actually commit history and the "files changed" tab.  :)